crates/telio-wg/src/link_detection/mod.rs

```diff
@@ -241,6 +294,28 @@ enum StateVariant {
     PossibleDown { deadline: Instant },
     Up,
 }
+
+struct StateUpdateResult {
+    should_ping: bool,
+    link_detection_update_result: LinkDetectionUpdateResult,
+}
+
+enum StateDecision {
+    NoAction,
+    Notify,
+    Ping,
+}
+
+impl StateDecision {
+    fn should_notify(&self) -> bool {
+        matches!(self, StateDecision::Notify)
+    }
+
+    fn should_ping(&self) -> bool {
+        matches!(self, StateDecision::Ping)
+    }
+}
```

Comment:
Hmm, not sure whether it is worth it to wrap this matches when they are used exactly once?